inc/api/namespace.php

```diff
@@ -1024,81 +809,84 @@ function get_post_diff_data( array $post_ids, $start, $end, $resolution = '1d' )
 			continue;
 		}
 
-		$value = $id;
-
 		// Determine attribute to filter results on.
 		if ( is_numeric( $id ) ) {
-			$term = 'attributes.postId.keyword';
-			$event = 'pageView';
 			if ( get_post_type( $id ) === 'wp_block' ) {
-				$term = 'attributes.blockId.keyword';
-				$event = 'blockView';
+				$block_view_ids[] = $id;
 			} elseif ( get_post_type( $id ) === 'xb' ) {
-				$term = 'attributes.clientId.keyword';
-				$event = 'experienceView';
-				$value = get_post( $id )->post_name;
+				$experience_view_ids[ $id ] = get_post( $id )->post_name;
+			} else {
+				$page_view_ids[] = $id;
 			}
 		} else {
-			$term = 'attributes.clientId.keyword';
-			$event = 'experienceView';
+			$experience_view_ids[] = $id;
 		}
-
-		// Add a top level aggregation.
-		$aggs['posts']['filters']['filters'][ $id ] = [
-			'bool' => [
-				'filter' => [
-					[ 'term' => [ $term => (string) $value ] ],
-					[ 'term' => [ 'event_type.keyword' => $event ] ],
-				],
-			],
-		];
 	}
 
-	if ( count( $post_ids ) > count( $data ) ) {
-		$results = [
-			'previous' => [],
-			'current' => [],
-		];
-
-		foreach ( array_keys( $results ) as $period ) {
-			$period_start = $period === 'previous' ? $start - $diff : $start;
-			$period_end = $period === 'previous' ? $start : $end;
-
-			// Set extended bounds.
-			$aggs['posts']['aggregations']['by_date']['date_histogram']['extended_bounds'] = [
-				'min' => (int) sprintf( '%d000', $period_start ),
-				'max' => (int) sprintf( '%d999', $period_end ),
-			];
+	$page_ids_sql = $wpdb->prepare( implode( ',', array_fill( 0, count( $page_view_ids ), '%s' ) ), ...$page_view_ids );
+	$block_ids_sql = $wpdb->prepare( implode( ',', array_fill( 0, count( $block_view_ids ), '%s' ) ), ...$block_view_ids );
+	$experience_ids_sql = $wpdb->prepare( implode( ',', array_fill( 0, count( $experience_view_ids ), '%s' ) ), ...$experience_view_ids );
 
-			$query = get_query(
-				[ 'pageView', 'experienceView', 'blockView' ],
-				$period_start,
-				$period_end,
-				$aggs
-			);
+	$resolution = esc_sql( $resolution );
 
-			$res = Utils\query( $query, [
-				'preference' => __FUNCTION__,
-			] );
+	// Generate fields for histogram of current date range.
+	$buckets = [];
+	$bucket_start = $start;
+	while ( $bucket_start < $end ) {
+		$bucket_end = strtotime( "+{$resolution}", $bucket_start );
+		$buckets[ $bucket_start ] = sprintf(
+			'uniqCombined64If(endpoint_id, event_timestamp >= toDateTime64(%1$d,3) AND event_timestamp < toDateTime64(%2$d,3)) as `%1$d`',
```

Comment:
This looks a bit weird, particularly the last `as` bit:
```
as `%1$d`
```

Reply:
I'm naming the field selected as the timestamp for the start of the histogram bucket, makes the post-processing very minimal.